ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/CriterionIndicator.java

```diff
@@ -0,0 +1,124 @@
+    private final Position position;
+
+    /**
+     * Constructor.
+     * 
+     * @param series                 the bar series
+     * @param tradingRecord          the trading record
+     * @param criterion              the criterion to get the calculated criterion
+     *                               value on a bar index
+     * @param requiredCriterionValue the required criterion value to test against
+     *                               the calculated criterion value
+     */
+    public CriterionIndicator(BarSeries series, TradingRecord tradingRecord, AnalysisCriterion criterion,
+            Num requiredCriterionValue) {
+        super(series);
+        this.tradingRecord = tradingRecord;
+        this.criterion = criterion;
+        this.requiredCriterionValue = requiredCriterionValue;
+        this.position = null;
+    }
+
+    /**
+     * Constructor.
+     * 
+     * @param series                 the bar series
+     * @param position               the position
+     * @param criterion              the criterion to get the calculated criterion
+     *                               value on a bar index
+     * @param requiredCriterionValue the required criterion value to test against
+     *                               the calculated criterion value
+     */
+    public CriterionIndicator(BarSeries series, Position position, AnalysisCriterion criterion,
+            Num requiredCriterionValue) {
+        super(series);
+        this.tradingRecord = null;
+        this.criterion = criterion;
+        this.requiredCriterionValue = requiredCriterionValue;
+        this.position = position;
+    }
+
+    @Override
+    protected Boolean calculate(int index) {
+        if (tradingRecord != null && !tradingRecord.getPositions().isEmpty()) {
+            TradingRecord tradingRecordTillIndex = new BaseTradingRecord(tradingRecord.getStartingType(),
+                    tradingRecord.getTransactionCostModel(), tradingRecord.getHoldingCostModel());
+            for (Position pos : tradingRecord.getPositions()) {
+                // consider only those positions made till index
+                if (pos.getEntry().getIndex() <= index) {
+                    Trade entry = pos.getEntry();
+                    tradingRecordTillIndex.enter(entry.getIndex(), entry.getNetPrice(), entry.getAmount());
+                }
+                if (pos.getExit() != null && pos.getExit().getIndex() <= index) {
+                    Trade exit = pos.getExit();
+                    tradingRecordTillIndex.exit(exit.getIndex(), exit.getNetPrice(), exit.getAmount());
+                }
+            }
+
+            Num calculatedCriterionValue = criterion.calculate(getBarSeries(), tradingRecordTillIndex);
+            return criterion.betterThan(calculatedCriterionValue, requiredCriterionValue)
+                    || calculatedCriterionValue.isEqual(requiredCriterionValue);
+
+        } else if (position != null) {
```

Comment:
I would not support to provide a `Position` or a `TradingRecord` in the construction. This check needs to be done in every calculation, but is always the same then

Reply:
>I would not support to provide a Position or a TradingRecord in the construction...but is always the same then

But the `calculate()` can start with an empty tradingRecord..which is filled while running the indicator. That's the reason for the check on every calculate()-method. Or do you mean something else?

Reply:
where does the TradingRecord gets filled? If you use the second constructor the TradinRecord is always `null`. I am also unsure if there is a real use case for this kind of indicators

Reply:
The point of this `CriterionIndicator` is: 
To have the ability to measure the `TradingRecord` while trading, the same as we measure the `BarSeries` with Indicators while trading. It's similar to https://github.com/ta4j/ta4j/pull/757, but more general - because you can put any kind of criterion within this `CriterionIndicator`. For example, you measure the PnL of the TradingRecord with this indicator (by `CriterionIndicator(new ProfitLossCriterion()`) to get a signal when to exit. Currently, we have no ability to use all the criterions as if there where indicators. 

While the typical Indicators only use `BarSeries` for its decision, with the CriterionIndicator it can use any `TradingRecord` for its decision.

I simplified the CriterionIndicator and removed the Position from the Constructor (it does not make sense to measure a Position, it would be the same as I would have a BarSeries with only one Bar).